library/src/main/java/com/google/android/exoplayer2/source/chunk/ChunkSampleStream.java

```diff
@@ -115,14 +115,29 @@ public long getBufferedPositionUs() {
   }
 
   /**
+   * Check if the input position locates within the chunk we are downloading now.
+   *
+   * @param positionUs The target position in microseconds.
+   * @return Whether the input position locates within the chunk we are downloading now.
+   */
+  public boolean isWithinLastChunk(long positionUs) {
+    if (isPendingReset() || loadingFinished) {
+      return false;
+    } else {
+      return ((positionUs >= getCurrentLoadPositionUs()) && (positionUs < getNextLoadPositionUs()));
+    }
+  }
+
+  /**
    * Seeks to the specified position in microseconds.
    *
    * @param positionUs The seek position in microseconds.
    */
   public void seekToUs(long positionUs) {
     lastSeekPositionUs = positionUs;
     // If we're not pending a reset, see if we can seek within the sample queue.
-    boolean seekInsideBuffer = !isPendingReset() && sampleQueue.skipToKeyframeBefore(positionUs);
+    boolean seekInsideBuffer = !isPendingReset() && 
+        sampleQueue.skipToKeyframeBefore(positionUs, isWithinLastChunk(positionUs));
```

Comment:
I'm not sure you need to be as sophisticated as checking whether the position is within the last chunk. Isn't it sufficient just to pass `positionUs < getNextLoadPositionUs()` as the second argument? I don't think passing true does any harm in cases where `positionUs` is *earlier* than the last chunk.

This allows you to remove your `isWithinLastChunk` and `getCurrentLoadPosition` methods ;).